scalyr_agent/copying_manager.py

```diff
@@ -846,6 +856,13 @@ def run(self):
 
                             if result == "success":
                                 last_success = current_time
+
+                            # Rate limit based on amount of copied log bytes in a successful request
+                            if self.__rate_limiter:
+                                self.__rate_limiter.block_until_charge_succeeds(
```

Comment:
I know there are tests for the rate limiter class itself in #543, but it would also be good to have some tests which cover that scenario for copying manager as well.

Reply:
And ideally, we would also have end to end test for it - e.g. we would set ``max_send_rate_enforcement`` to some low value and then we would try to consume some large-ish file and make sure it's eventually consumed (taking into account sleep, etc.).

Reply:
Due to time constraints on this release and me personally not seeing any easy to add tests for this the plan is to leave these for later. If anyone has suggestions for easy to add tests I will gladly take them though.

Reply:
Alright, let's then please get back to this once the release is out.